pkg/cloud/vsphere/actuators/cluster/actuator.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"github.com/pkg/errors"
+
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/klogr"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	clientv1 "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/typed/cluster/v1alpha1"
+
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/context"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/services/certificates"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/utils"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/record"
+)
+
+// Actuator is responsible for maintaining the Cluster objects.
+type Actuator struct {
+	client     clientv1.ClusterV1alpha1Interface
+	coreClient corev1.CoreV1Interface
+}
+
+// NewActuator returns a new instance of Actuator.
+func NewActuator(
+	client clientv1.ClusterV1alpha1Interface,
+	coreClient corev1.CoreV1Interface) *Actuator {
+
+	return &Actuator{
+		client:     client,
+		coreClient: coreClient,
+	}
+}
+
+// Reconcile will create or update the cluster
+func (a *Actuator) Reconcile(cluster *clusterv1.Cluster) (result error) {
```

Comment:
nit: `result error` is a little confusing for me, I would expect `result` to be an object and error to be named `e` or `err` ?

Reply:
Hi @moshloop,

It's because if you use `err`, you can't use `err` in the function where `err` may be returned from a tuple, and I don't like not being able to use that. The only reason I use a named return param is to be able to set or check it during `defer` logic.

Reply:
hmm starting to see the benefits for consistency though

Reply:
Hi @andrewsykim and @moshloop,

There's only one error case *for now*. That will change as soon as we introduce downloading the OVA into the content library, VMC LB support, etc. I'm leaving this as is as it's consistent with the other functions and will be needed in not even weeks.